modules/onewire/onewire.c

```diff
@@ -0,0 +1,418 @@
+    for (i = 0; i < ROM_DATA_SIZE_BYTES; i++)
+    rom_backup[i] = OneWireStruct->ROM_NO[i];
+    ld_backup = OneWireStruct->LastDiscrepancy;
+    ldf_backup = OneWireStruct->LastDeviceFlag;
+    lfd_backup = OneWireStruct->LastFamilyDiscrepancy;
+
+    /* Set search to find the same device */
+    OneWireStruct->LastDiscrepancy = ROM_DATA_SIZE_BITS;
+    OneWireStruct->LastDeviceFlag = 0;
+
+    if (OneWire_Search(OneWireStruct, ONEWIRE_CMD_SEARCHROM)) {
+        /* Check if same device found */
+        rslt = 1;
+        for (i = 0; i < ROM_DATA_SIZE_BYTES; i++) {
+            if (rom_backup[i] != OneWireStruct->ROM_NO[i]) {
+                rslt = 1;
+                break;
+            }
+        }
+    } else {
+        rslt = 0;
+    }
+
+    /* Restore the search state */
+    for (i = 0; i < ROM_DATA_SIZE_BYTES; i++) {
+        OneWireStruct->ROM_NO[i] = rom_backup[i];
+    }
+    OneWireStruct->LastDiscrepancy = ld_backup;
+    OneWireStruct->LastDeviceFlag = ldf_backup;
+    OneWireStruct->LastFamilyDiscrepancy = lfd_backup;
+
+    /* Return the result of the verify */
+    return rslt;
+}
+
+uint8_t OneWire_TargetSetup(OneWire_t* OneWireStruct, uint8_t family_code) {
+    if (!OneWireStruct) return 0;
+    uint8_t i;
+
+    /* Set the search state to find SearchFamily type devices */
+    OneWireStruct->ROM_NO[0] = family_code;
+    for (i = 1; i < ROM_DATA_SIZE_BYTES; i++) {
+        OneWireStruct->ROM_NO[i] = 0;
+    }
+
+    OneWireStruct->LastDiscrepancy = ROM_DATA_SIZE_BITS;
+    OneWireStruct->LastFamilyDiscrepancy = 0;
+    OneWireStruct->LastDeviceFlag = 0;
+    return 1;
+}
+
+uint8_t OneWire_FamilySkipSetup(OneWire_t* OneWireStruct) {
+    if (!OneWireStruct) return 0;
+    /* Set the Last discrepancy to last family discrepancy */
+    OneWireStruct->LastDiscrepancy = OneWireStruct->LastFamilyDiscrepancy;
+    OneWireStruct->LastFamilyDiscrepancy = 0;
+
+    /* Check for end of list */
+    if (OneWireStruct->LastDiscrepancy == 0) {
+        OneWireStruct->LastDeviceFlag = 1;
+    }
+    return 1;
+}
+
+char OneWire_GetROM(OneWire_t* OneWireStruct, uint8_t index) {
+    if (!OneWireStruct)                           return -1;
+    if (index > 0 || index < ROM_DATA_SIZE_BYTES) return -1;
+    return OneWireStruct->ROM_NO[index];
+}
+
+uint8_t OneWire_Select(OneWire_t* OneWireStruct, uint8_t* addr) {
+    if (!OneWireStruct) return 0;
+    if (!addr)          return 0;
+    uint8_t i;
+    OneWire_WriteByte(OneWireStruct, ONEWIRE_CMD_MATCHROM);
+    
+    for (i = 0; i < ROM_DATA_SIZE_BYTES; i++) {
+        OneWire_WriteByte(OneWireStruct, *(addr + i));
+    }
+    return 1;
+}
+
+uint8_t OneWire_SelectWithPointer(OneWire_t* OneWireStruct, uint8_t *ROM) {
+    if (!OneWireStruct) return 0;
+    if (!ROM)           return 0;
+    uint8_t i;
+    OneWire_WriteByte(OneWireStruct, ONEWIRE_CMD_MATCHROM);
+    
+    for (i = 0; i < ROM_DATA_SIZE_BYTES; i++) {
+        OneWire_WriteByte(OneWireStruct, *(ROM + i));
+    }   

```

Comment:
Same commend as a few above. Let's nix `uint8_t i;` and _consider_ the square bracket operator.